### PR TITLE
[topic-gpio] samples: led_apa102c_bitbang: Convert to new GPIO API

### PR DESCRIPTION
**Pull request by carlescufi**
Conver to the new GPIO API, using raw access since there is no DT
definitions for this particular usage of the pins.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

### CODE REVIEW

File: samples/drivers/led_apa102c_bitbang/src/main.c [hunk 1]
```diff
@@ -79,12 +79,12 @@ void main(void)
 	}
 
 	/* Setup GPIO output */
-	ret = gpio_pin_configure(gpio_dev, GPIO_DATA_PIN, (GPIO_DIR_OUT));
+	ret = gpio_pin_configure(gpio_dev, GPIO_DATA_PIN, (GPIO_OUTPUT));
```
Comment (pabigot) [inline samples/drivers/led_apa102c_bitbang/src/main.c]: There's really no reason why that parameter needs to be in parentheses, but changing it isn't necessary.

**Comment by carlescufi** [comment]
Untested, I don't have the hardware.

File: samples/drivers/led_apa102c_bitbang/src/main.c [hunk 2]
```diff
@@ -55,11 +55,11 @@ void send_rgb(struct device *gpio_dev, u32_t rgb)
 
 	for (i = 0; i < 32; i++) {
 		/* MSB goes in first */
-		gpio_pin_write(gpio_dev, GPIO_DATA_PIN, !!(rgb & 0x80000000));
+		gpio_pin_set_raw(gpio_dev, GPIO_DATA_PIN, !!(rgb & 0x80000000));
```
Comment (pabigot) [inline samples/drivers/led_apa102c_bitbang/src/main.c]: MISRA-inclined folks would prefer this to change to `(rgb & 0x80000000U) != 0` or (better) `(rgb & BIT(31)) != 0`.

**Review by pabigot — APPROVED** [review]
Possible cleanups but this should be fine.

File: samples/drivers/led_apa102c_bitbang/src/main.c (hunk 1 above)
Comment (carlescufi) [reply samples/drivers/led_apa102c_bitbang/src/main.c]: Fixed, thanks

File: samples/drivers/led_apa102c_bitbang/src/main.c (hunk 2 above)
Comment (carlescufi) [reply samples/drivers/led_apa102c_bitbang/src/main.c]: Why not, since we are touching this file. Fixed, thanks.

**Review by mnkp — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**